### PR TITLE
Catch wrong datetime and prevent fatals [Delivers #90401482]

### PR DESCRIPTION
**Pull request by Petervw**

### CODE REVIEW

File: classes/class-sitemap.php [hunk 1]
```diff
@@ -87,9 +87,10 @@ public function build() {
 					}
 				}
 
-				$keywords      = new WPSEO_News_Meta_Keywords( $item->ID );
-				$genre         = $this->get_item_genre( $item->ID );
-				$stock_tickers = $this->get_item_stock_tickers( $item->ID );
+				$keywords         = new WPSEO_News_Meta_Keywords( $item->ID );
+				$genre            = $this->get_item_genre( $item->ID );
+				$stock_tickers    = $this->get_item_stock_tickers( $item->ID );
+				$publication_date = $this->get_publication_date( $item->post_date_gmt );
```
Comment (omarreiss) [inline classes/class-sitemap.php]: Please pass in the `$item` here and make sure you check the `$item->post_date`, `$item->post_modified_gmt` and `$item->post_modified` before you fallback to an empty string.

**Comment by omarreiss** [comment]
Please fix codestyle

**Comment by omarreiss** [comment]
@Petervw CR 1 finished.

File: classes/class-sitemap.php [hunk 2]
```diff
@@ -207,10 +208,26 @@ private function get_publication_date( $item_date ) {
 			$timezone_string = $this->wp_get_timezone_string();
 		}
 
-		// Create a DateTime object date in the correct timezone
-		$datetime = new DateTime( $item_date, new DateTimeZone( $timezone_string ) );
+		if ( $this->is_valid_datetime( $item->post_date_gmt ) ) {
+			// Create a DateTime object date in the correct timezone
+			$datetime = new DateTime( $item->post_date_gmt, new DateTimeZone( $timezone_string ) );
```
Comment (andizer) [inline classes/class-sitemap.php]: Can you DRY this, because I see three time the same kind of code .

**Comment by omarreiss** [comment]
CR :+1:

File: classes/class-sitemap.php [hunk 3]
```diff
@@ -207,8 +208,32 @@ private function get_publication_date( $item_date ) {
 			$timezone_string = $this->wp_get_timezone_string();
 		}
 
-		// Create a DateTime object date in the correct timezone
-		$datetime = new DateTime( $item_date, new DateTimeZone( $timezone_string ) );
+		if ( $this->is_valid_datetime( $item->post_date_gmt ) ) {
+			// Create a DateTime object date in the correct timezone
+			return $this->format_date_from_timezone( $item->post_date_gmt, $timezone_string );
+		}
+		elseif ( $this->is_valid_datetime( $item->post_modified_gmt ) ) {
```
Comment (omarreiss) [inline classes/class-sitemap.php]: @Petervw Whoops you forgot the `$item->post_date`

**Comment by andizer** [comment]
Awesome!

**Comment by omarreiss** [comment]
Joint CR :+1:

**Comment by Petervw** [comment]
:100: points! :+1:

**Comment by andizer** [comment]
Acceptance test done, wrapper works and when all given dates are invalid the string is empty. Ready to merge.

**Pull request merged**